### ISSUE TITLE
CPU Temperature Sensor

### ISSUE THREAD

**Issue by tessierp** [LibreHardwareMonitor/LibreHardwareMonitor#1577]
Trying to get the my 7800X3D temperature sensor. Only 1 is returned and the value is always 0 which I find very odd. Could that be a bug? I tried the application that you guys made and temps are shown so not sure if this is a bug or perhaps you calculate it manually?

This is all I get in terms of sensors which leads me to believe there might be a bug : 

![image](https://github.com/user-attachments/assets/de689e6b-4380-418f-8010-cd01c85ccbf3)

**Comment by jansteffen** [comment]
5800X3D here, it's not showing me CPU temps either:

![image](https://github.com/user-attachments/assets/df6ce228-a2c7-40ec-8ab2-522c2cca85ad)

**Comment by sebastian-dev** [comment]
This is correct, the cpu has one temperature sensor. Additional sensors are only available via SMU.

**Comment by tessierp** [comment]
> This is correct, the cpu has one temperature sensor. Additional sensors are only available via SMU.

I'm not sure I understand.. I'm looking for the sensor type temperature, what is returned to me is a value of 0 which isn't right.

**Comment by sebastian-dev** [comment]
Post a Report (File).
It looks like the driver is not installed correct. Maybe blocked from a AV soulution.

**Comment by sebastian-dev** [comment]
Application (Menu) File->Save Report.
Attach this file here.

**Comment by tessierp** [comment]
> Application (Menu) File->Save Report. Attach this file here.

But wait I'm not using the application, I'm using the API to output data to ZABBIX and as I look through all the available sensors and the data that comes back has no temperature sensor.

**Comment by tessierp** [comment]
Hello? Any clue anyone?

**Comment by sebastian-dev** [comment]
Use the application (LHM) and check whether the data is available there.
You can also create a report from the application.
If there are no problems, the problem lies with the other software.

**Comment by tessierp** [comment]
> Use the application (LHM) and check whether the data is available there. You can also create a report from the application. If there are no problems, the problem lies with the other software.

As I explained, I'm using HWINFO, AMD Adrenaline as well as many others they all report the temperatures.. I also tried the application made by your team and yes the temperature is displayed.  But that is NOT THE ISSUE. The issue is when I use the LibreHardwareMonitor's libraries to crawl through all the sensors available, the temperature sensor is not in the list!! Someone else also reported that LHM is not reporting temps.

So I guess I'll have to show you what I am doing.

First I get the details of my CPU 

```
        static IEnumerable<CpuData> GetSystemCPUInfo()
        {
            Computer computer = OpenComputer();

            try
            {
                computer.Accept(new UpdateVisitor());

                var cpuData =
                    (from hw in computer.Hardware
                     where hw.HardwareType == HardwareType.Cpu
                     select new CpuData(hw)).ToList(); ;

                return cpuData;
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.ToString());
                throw;
            }
        }
```

Then, from the data that is returned I enumerate all the sensors

```
                var cpuDataTest = GetSystemCPUInfo();
                foreach (ISensor sensor in cpuDataTest.FirstOrDefault().Sensors)
                {
                    Console.WriteLine($"\tSensor: {sensor.Name}, type {sensor.SensorType.ToString()}, value: {sensor.Value}");
                }
```

and the list of sensors I already gave you from what I wrote above. So my question again is WHY is the temperature sensor not shown?

**Comment by Data-Corruption** [comment]
Hello, I had the same issue, running my exe as admin fixed it, hope it helps.

**Comment by tessierp** [comment]
> Hello, I had the same issue, running my exe as admin fixed it, hope it helps.

Well that fixed my issue, running Jetbrain Rider in admin and debugging the application returned a value for the temperature sensor! 

Thanks @Data-Corruption !
